packages/gateway/src/features/user/type-resolvers.ts

```diff
@@ -221,32 +221,42 @@ export const userTypeResolvers: GQLResolver = {
 
       const signatureExtension = getSignatureExtension(practitioner.extension)
 
-      const signature =
+      const presignedUrl =
         userModel.systemRole === 'FIELD_AGENT'
           ? null
-          : signatureExtension && signatureExtension.valueSignature
-
+          : signatureExtension &&
+            (await getPresignedUrlFromUri(
+              signatureExtension.valueAttachment.url,
+              authHeader
+            ))
       return {
         role: practitionerRole,
         name: practitioner.name,
-        signature: signature && {
-          type: signature.contentType,
-          data: signature.blob
-        }
+        signature: presignedUrl
       }
     },
-    async signature(userModel: IUserModelData, _, { dataSources }) {
+    async signature(
+      userModel: IUserModelData,
+      _,
+      { headers: authHeader, dataSources }
+    ) {
       const practitioner = await dataSources.fhirAPI.getPractitioner(
         userModel.practitionerId
       )
 
       const signatureExtension = getSignatureExtension(practitioner.extension)
 
-      const signature = signatureExtension && signatureExtension.valueSignature
+      const presignedUrl =
+        signatureExtension &&
+        (await getPresignedUrlFromUri(
+          signatureExtension.valueAttachment.url,
+          authHeader
+        ))
+
       return (
-        signature && {
-          type: signature.contentType,
-          data: signature.blob
+        presignedUrl && {
+          type: 'minioUrl',
```

Comment:
```suggestion
          type: signatureExtension?.contentType,
```
Or better yet, let's do an early return if the signatureExtension is not there